libs/search-widget/src/viewer/viewer.store.ts

```diff
@@ -184,6 +211,25 @@ export function clearSearch() {
   viewerStore.onlySelected.next(false);
 }
 
+export function setParagraphLabels(labels: Classification[], paragraph: WidgetParagraph) {
+  const resource = viewerStore.resource.getValue();
+  const saving = viewerStore.savingLabels.getValue();
+  if (!resource || saving) return;
+  viewerStore.savingLabels.next(true);
+  const paragraphId = getParagraphId(resource.id, paragraph);
+  setLabels(resource, paragraph.fieldId, paragraph.fieldType.slice(0, -1), paragraphId, labels)
```

Comment:
`paragraph.fieldType.slice(0, -1)` I don't think this is correct as first letter of fieldType is not always the same as the one letter expected by the backend. For example `link` is `u`.
We know we have only one field type and field id for all paragraphs displayed on the viewer currently (because we don't display several fields at once in the viewer), so we stored the current field in the store for entities: `viewerStore.currentField`

Reply:
Fixed